From e917e3111b2d2b27db21b47c9586d231c5785be5 Mon Sep 17 00:00:00 2001
From: "Carsten Haitzler (Rasterman)" <raster@rasterman.com>
Date: Tue, 26 May 2020 12:16:09 +0100
Subject: [PATCH 1/7] evas - csd and wl csd - fix alpha zeroing to avoid
 framespace junk

i noticed some junk in framespace... this happesn when solid objects
are outside the normal window content area and their solid areas are
not clipped to this inner frame area thus those areas are not "zeroed
out" next render.

@fix
---
 src/lib/evas/canvas/evas_render.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git src/lib/evas/canvas/evas_render.c src/lib/evas/canvas/evas_render.c
index 1d47ba06af..567461d1dd 100644
--- src/lib/evas/canvas/evas_render.c
+++ src/lib/evas/canvas/evas_render.c
@@ -2722,6 +2722,15 @@ _evas_render_cutout_add(Evas_Public_Data *evas, void *context,
                            obj->cur->cache.clip.w, obj->cur->cache.clip.h);
      }
    else return;
+   if (!_is_obj_in_framespace(obj, evas))
+     {
+        int fw, fh;
+
+        fw = evas->viewport.w - evas->framespace.w;
+        fh = evas->viewport.h - evas->framespace.h;
+        RECTS_CLIP_TO_RECT(cox, coy, cow, coh,
+                           0, 0, fw, fh);
+     }
    if (cutout_margin)
      {
         cox += cutout_margin->l;
-- 
2.27.0

